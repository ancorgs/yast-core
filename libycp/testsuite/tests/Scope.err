Parsed:
----------------------------------------------------------------------
{
  integer `DoubleUsedVariable = 0;
  integer `Loop = 0;
  while ((Loop < 5)) {
  integer `DoubleUsedVariable = 5;
  `DoubleUsedVariable = ((DoubleUsedVariable * DoubleUsedVariable));
  `Loop = ((Loop + 1));
}
  `DoubleUsedVariable = ((DoubleUsedVariable - 2));
  while ((Loop < 5)) {
  integer `DoubleUsedVariable = 5;
  `DoubleUsedVariable = ((DoubleUsedVariable * DoubleUsedVariable));
  `Loop = ((Loop + 1));
}
  `DoubleUsedVariable = ((DoubleUsedVariable * DoubleUsedVariable));
  return DoubleUsedVariable;
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
  any `MultipleUsedVariable = 0;
  if (true) {
  any `MultipleUsedVariable = 1;
  `MultipleUsedVariable = ((MultipleUsedVariable + 1));
  if (true) {
  `MultipleUsedVariable = ("I am a string");
}
}
  return MultipleUsedVariable;
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
  any `MultipleUsedVariable = 0;
  if (true) {
  any `MultipleUsedVariable = 1;
  `MultipleUsedVariable = ((MultipleUsedVariable + 1));
  if (true) {
  while (true) {
  string `MultipleUsedVariable = "";
  `MultipleUsedVariable = ("I am a string");
  break;
}
  `MultipleUsedVariable = (1.20000000000000);
}
}
  return MultipleUsedVariable;
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
  any `MultipleUsedVariable = 0;
  if (true) {
  any `MultipleUsedVariable = 1;
  `MultipleUsedVariable = ((MultipleUsedVariable + 1));
  if (true) {
  while (true) {
  string `MultipleUsedVariable = "";
  `MultipleUsedVariable = ("I am a string");
  return MultipleUsedVariable;
}
  `MultipleUsedVariable = (1.20000000000000);
}
}
  return MultipleUsedVariable;
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
  any `MultipleUsedVariable = 0;
  if (true) {
  any `MultipleUsedVariable = 1;
  `MultipleUsedVariable = ({
  while (true) {
  string `MultipleUsedVariable = "";
  `MultipleUsedVariable = ("I am a string");
  return MultipleUsedVariable;
}
}
);
}
  return MultipleUsedVariable;
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
  integer `a = 1;
  integer `b = 2;
  integer `c = 3;
  if (true) {
  integer `c = 5;
  undefine c;
  `c = (27);
}
  return c;
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
  integer `first = 1;
  string `first = "";
  float `first = 0.00000000000000;
  return first;
}
----------------------------------------------------------------------
[Interpreter] tests/Scope.ycp:169 Symbol 'first' already declared at line 168
[Interpreter] tests/Scope.ycp:170 Symbol 'first' already declared at line 169
