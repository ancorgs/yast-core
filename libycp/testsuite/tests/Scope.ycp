
# ---------------------------------------------------------
#
#  Filename:    Scope.ycp
#
#  Purpose:     test cases for YCP test
#
#  Creator:     jd@suse.de
#
#  Maintainer:  jd@suse.de
#
# ---------------------------------------------------------


// this tests includes tests for the scope of variables


// the variable is declared twice
// the loop is defined, because there is a declaration within that block
{

   integer DoubleUsedVariable = 0;
   integer Loop = 0;

   while( Loop < 5 )
   {
       integer DoubleUsedVariable = 5;

       DoubleUsedVariable = DoubleUsedVariable * DoubleUsedVariable;
       Loop = Loop + 1;
   }

   DoubleUsedVariable = DoubleUsedVariable - 2;

   while( Loop < 5 )
   {
       integer DoubleUsedVariable = 5;

       DoubleUsedVariable = DoubleUsedVariable * DoubleUsedVariable;
       Loop = Loop + 1;
   }

   DoubleUsedVariable = DoubleUsedVariable * DoubleUsedVariable;

   return DoubleUsedVariable;
}

// the variable is declared multiple
// the block is left with the last return statement
{

   any MultipleUsedVariable = 0;

   if( true )
   {
       any MultipleUsedVariable = 1;

       MultipleUsedVariable = MultipleUsedVariable + 1;
       if( true )
       {
           MultipleUsedVariable = "I am a string";
       }
   }

   return MultipleUsedVariable;
}

// the variable is declared multiple
// the block is left with the last return statement
// the multiple declared variable is declared with different types
{

   any MultipleUsedVariable = 0;

   if( true )
   {
       any MultipleUsedVariable = 1;

       MultipleUsedVariable = MultipleUsedVariable + 1;
       if( true )
       {
           while( true )
           {
               string MultipleUsedVariable = "";

               MultipleUsedVariable = "I am a string";
               break;
           } 
           MultipleUsedVariable = 1.20;
       }
   }

   return MultipleUsedVariable;
}

// the variable is declared multiple
// an inner block is left with return without assignment
{

   any MultipleUsedVariable = 0;

   if( true )
   {
       any MultipleUsedVariable = 1;

       MultipleUsedVariable = MultipleUsedVariable + 1;
       if( true )
       {
           while( true )
           {       
               string MultipleUsedVariable = "";

               MultipleUsedVariable = "I am a string";
               return MultipleUsedVariable;
           }       
           MultipleUsedVariable = 1.20; 
       }
   }

   return MultipleUsedVariable;
}

// the variable is declared multiple
// an inner block is left with return without assignment
{

   any MultipleUsedVariable = 0;

   if( true )
   {
       any MultipleUsedVariable = 1;

       MultipleUsedVariable = 
       {
           while( true )
           {
               string MultipleUsedVariable = "";

               MultipleUsedVariable = "I am a string";
               return MultipleUsedVariable;
           }
       };
   }

   return MultipleUsedVariable;
}


// check undefine
{
  integer a = 1;
  integer b = 2;
  integer c = 3;

  if (true) {
    integer c = 5;

    undefine c;

    c = 27;
  }

  return c;
}

// double declaration
{
  integer first = 1;
  string first = "";
  float first = 0.0;

  return first;
}
