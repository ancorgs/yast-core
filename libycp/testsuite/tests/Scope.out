(4)
(0)
(0)
("I am a string")
(0)
(27)
(0.00000000000000)
